mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp

```diff
@@ -89,8 +89,8 @@ struct SqrtReciprocalOptimization : public OpRewritePattern<tosa::PowOp> {
 
     auto constantType = scale.getElementType();
     float scaleValue = 0.;
-    if (constantType.isF32())
-      scaleValue = scale.getSplatValue<float>();
+    if (constantType.isF32() || constantType.isBF16())
```

Comment:
Would it be a problem to also add fp16? Pow and rsqrt also support fp16 and then that would mean we cover all potential uses cases.

Reply:
I agree with Chris. We don't need to check for the type here. We can support any type (since all are FP types).

Reply:
@flemairen6 could you also check we are not also restricting to F32 in other canonicalizations?